app/src/androidTest/java/ch/epfl/sdp/drone3d/ui/weather/WeatherInfoActivityTest.kt

```diff
@@ -0,0 +1,117 @@
+import ch.epfl.sdp.drone3d.service.api.weather.WeatherService
+import ch.epfl.sdp.drone3d.service.module.WeatherModule
+import ch.epfl.sdp.drone3d.ui.map.MissionInProgressActivity
+import ch.epfl.sdp.drone3d.ui.mission.ItineraryShowActivity
+import com.mapbox.mapboxsdk.geometry.LatLng
+import dagger.hilt.android.testing.BindValue
+import dagger.hilt.android.testing.HiltAndroidRule
+import dagger.hilt.android.testing.HiltAndroidTest
+import dagger.hilt.android.testing.UninstallModules
+import org.hamcrest.Matchers.not
+import org.junit.*
+import org.junit.rules.RuleChain
+import org.mockito.Mockito
+import org.mockito.Mockito.`when`
+import java.util.*
+
+/**
+ * Test WeatherInfoActivity
+ */
+@HiltAndroidTest
+@UninstallModules(WeatherModule::class)
+class WeatherInfoActivityTest {
+    companion object {
+        private val GOOD_VISIBILITY_REPORT = WeatherReport("Clear", "description",
+            20.0, 20, 5.0, 10000, Date(12903))
+
+        private val BAD_VISIBILITY_REPORT = WeatherReport("RAIN", "description",
+            -1.0, 20, 10.0, 100, Date(12903))
+
+        private val someLocationsList = arrayListOf(
+            LatLng(47.398979, 8.543434),
+            LatLng(47.398279, 8.543934),
+            LatLng(47.397426, 8.544867),
+            LatLng(47.397026, 8.543067)
+        )
+    }
+
+    private val activityRule = ActivityScenarioRule<MissionInProgressActivity>(
+        Intent(ApplicationProvider.getApplicationContext(), WeatherInfoActivity::class.java).apply {
+            putExtras(
+                Bundle().apply {
+                    putSerializable(ItineraryShowActivity.FLIGHTPATH_INTENT_PATH, someLocationsList)
+                }
+            )
+        }
+    )
+
+    @get:Rule
+    val ruleChain: RuleChain = RuleChain.outerRule(HiltAndroidRule(this))
+        .around(activityRule)
+
+    @BindValue
+    val weatherService: WeatherService = Mockito.mock(WeatherService::class.java)
+
+    init {
+        `when`(weatherService.getWeatherReport(someLocationsList[0])).thenReturn(MutableLiveData(GOOD_VISIBILITY_REPORT))
+    }
+
+    @Before
+    fun setUp() {
+        Intents.init()
+    }
+
+    @After
+    fun tearDown() {
+        Intents.release()
+    }
+
+    /**
+     * Make sure the context of the app is the right one
+     */
+    @Test
+    fun useAppContext() {
+        // Context of the app under test.
+        val appContext = InstrumentationRegistry.getInstrumentation().targetContext
+        Assert.assertEquals("ch.epfl.sdp.drone3d", appContext.packageName)
+    }
+
+    @Test
+    fun visibilityTextAndColorCorrectWhenGoodVisibility() {
+        `when`(weatherService.getWeatherReport(someLocationsList[0])).thenReturn(MutableLiveData(GOOD_VISIBILITY_REPORT))
+
+        activityRule.scenario.recreate()
+
+        onView(withId(R.id.infoVisibility)).check(matches(withText("Visibility limit : 10 km")))
+        onView(withId(R.id.infoVisibility)).check(matches(not(hasTextColor(R.color.red))))
```

Comment:
It sadly is not possible to check that the color is black, because the color is not black. It is the default color, and the default color isn't always the same depending on the phone.